include/loops/loops.hpp

```diff
@@ -783,5 +837,23 @@ template<typename _Tp> VReg<_Tp> pow(const VReg<_Tp>& a, int p)
     return ret;
 }
 
+struct exp_consts
+{
+    VReg<float> lo, hi, half, one, LOG2EF, C1, C2, p0, p1, p2, p3, p4, p5;
+    VReg<int32_t> _7f;
+    exp_consts(VReg<float>&& a_lo, VReg<float>&& a_hi, VReg<float>&& a_half, VReg<float>&& a_one,
+                        VReg<float>&& a_LOG2EF, VReg<float>&& a_C1, VReg<float>&& a_C2, VReg<float>&& a_p0,
+                        VReg<float>&& a_p1, VReg<float>&& a_p2, VReg<float>&& a_p3, VReg<float>&& a_p4,
+                        VReg<float>&& a_p5, VReg<int32_t>&& a_7f);
```

Comment:
maybe this constructor is not needed. There should also be default constructor, perhaps

Reply:
There cannot be default constructor, it's needed at least Context.
That's why this constructor was replaced with exp_consts(Context CTX);